apps/frontend/src/libs/components/router-provider/protected-route.tsx

```diff
@@ -0,0 +1,24 @@
+import { FC } from "react";
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+
+import { AppRoute } from "~/libs/enums/enums.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+const ProtectedRoute: FC = () => {
```

Comment:
Should receive children that need to be displayed. Also, add a prop `redirectTo`, which will be AppRoute